sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -303,15 +315,35 @@ public class ChatClient {
         event: ChatEventId,
         handler: @escaping TrouterEventHandler
     ) {
-        if signalingClient == nil {
+        guard let signalingClient = signalingClient else {
+            if event == ChatEventId.realTimeNotificationConnected {
+                realTimeNotificationConnectedHandler = handler
+                return
+            }
+
+            if event == ChatEventId.realTimeNotificationDisconnected {
+                realTimeNotificationDisconnectedHandler = handler
+                return
+            }
+
             options.logger
                 .warning(
-                    "Signaling client is not initialized, cannot register handler. Ensure startRealtimeNotifications() is called first."
+                    "Signaling client is not initialized, cannot register handler."
```

Comment:
Out of curiosity, does this actually have to be true? Couldn't you use the same pattern as with RT notification started/stopped and simply store the handlers on the ChatClient and then "connect" them once the RT notifications have started? 

It would be a non-breaking change to add this in the future, but worth considering.